core/src/main/scala/com/yahoo/maha/utils/GetTotalRowsRequest.scala

```diff
@@ -0,0 +1,96 @@
+    rowList.addRow(row)
+  }
+
+  def getQueryExecutorContext: QueryExecutorContext = {
+    val qeOracle = new QueryExecutor {
+      override def execute[T <: RowList](query: Query, rowList: T, queryAttributes: QueryAttributes) : QueryResult[T] = {
+        updateRowList(rowList)
+        QueryResult(rowList, queryAttributes, QueryResultStatus.SUCCESS)
+      }
+      override def engine: Engine = OracleEngine
+    }
+
+    val qeHive = new QueryExecutor {
+      override def execute[T <: RowList](query: Query, rowList: T, queryAttributes: QueryAttributes) : QueryResult[T] = {
+        updateRowList(rowList)
+        QueryResult(rowList, queryAttributes, QueryResultStatus.SUCCESS)      }
+      override def engine: Engine = HiveEngine
+    }
+    val qeDruid = new QueryExecutor {
+      override def execute[T <: RowList](query: Query, rowList: T, queryAttributes: QueryAttributes) : QueryResult[T] = {
+        updateRowList(rowList)
+        QueryResult(rowList, queryAttributes, QueryResultStatus.SUCCESS)      }
+      override def engine: Engine = DruidEngine
+    }
+
+    val qec = new QueryExecutorContext
+    qec.register(qeOracle)
+    qec.register(qeHive)
+    qec.register(qeDruid)
+    qec
+  }
+
+  def getTotalRowsRequest(request: ReportingRequest, pipeline: QueryPipeline) : Try[ReportingRequest] = {
+    //no filters except fk filters
+    Try {
+      require(
+        pipeline.bestDimCandidates.nonEmpty
+        , s"Invalid total rows request, no best dim candidates! : $request")
+
+      //force dim driven
+      //remove all fields except primary key
+      //remove all sorts
+      val primaryKeyAliasFields = pipeline.bestDimCandidates.map(dim => Field(dim.publicDim.primaryKeyByAlias, None, None)).toIndexedSeq
+      request.copy(
+        selectFields = primaryKeyAliasFields
+        , sortBy = IndexedSeq.empty
+        , includeRowCount = true
+        , forceDimensionDriven = true
+        , forceFactDriven = false
+        , paginationStartIndex = 0
+        , rowsPerPage = request.rowsPerPage
+      )
+    }
+  }
+
+  def getTotalRows(request: ReportingRequest, sourcePipeline: QueryPipeline, registry: Registry) : Try[Int] = {
+    Try {
+      val totalRowsRequest: ReportingRequest = getTotalRowsRequest(request, sourcePipeline).get
+      val model: RequestModel = RequestModel.from(totalRowsRequest, registry).get
+      val maxRows: Int = DruidQueryGenerator.defaultMaximumMaxRows
+      assert(model.maxRows <= maxRows, throw new Exception(s"Value of ${model.maxRows} exceeds posted limit of $maxRows"))
+      val queryContext: QueryExecutorContext = getQueryExecutorContext
+
+      val rowListAttempt = sourcePipeline.execute(queryContext)
+      assert(rowListAttempt.isSuccess, "Failed to get valid executor and row list")
+
+      //Can fail back in getValue exception.
+      var result = 0
+      rowListAttempt.get._1.foreach(input => {
+        result += input.aliasMap(OracleQueryGenerator.ROW_COUNT_ALIAS).toString.toInt
```

Comment:
Probably need to check the number of rows returned.  Add more debug logging if debug is enabled.

Reply:
Shouldn't we determine which engine the request was executed and based on that parse the result?  Or are we expecting all the engines to just use the same alias/name?

Reply:
Believe only Oracle has this alias, and all others have nothing at all.  I can add in a check to make sure it is running an Oracle query.